### ISSUE TITLE
Better performance with quantized input

### ISSUE THREAD

**Issue by emiliopaolini** [larq/larq#713]
Hi everyone, i was doing some experiments using Larq and DoReFa quantizer. One thing that i noticed is that if i quantize the inputs, i get better results than using floating input. Is it possible or is this some sort of a bug? In particular the code that i use is the following:
```python
(train_images, train_labels), (test_images, test_labels) = tf.keras.datasets.mnist.load_data()

train_images = train_images.reshape((60000, 28, 28, 1))
test_images = test_images.reshape((10000, 28, 28, 1))


train_images, test_images = train_images / 255, test_images / 255
bitwidth = 4
kwargs = dict(input_quantizer=lq.quantizers.DoReFa(k_bit=bitwidth, mode="activations"),
              kernel_quantizer=lq.quantizers.DoReFa(k_bit=bitwidth, mode="weights"),
              kernel_constraint="weight_clip",
              activation="relu"
              )

model = tf.keras.models.Sequential()

size = (3,3)

model.add(lq.layers.QuantConv2D(32, 
                                kernel_size = size,
                                kernel_quantizer=lq.quantizers.DoReFa(k_bit=bitwidth, mode="weights"),
                                kernel_constraint="weight_clip",
                                #input_quantizer=lq.quantizers.DoReFa(k_bit=bitwidth, mode="activations"),
                                use_bias=False,
                                #activation="relu",
                                input_shape=(28, 28, 1)))
model.add(tf.keras.layers.MaxPooling2D((2, 2)))
model.add(tf.keras.layers.BatchNormalization(scale=False))

model.add(lq.layers.QuantConv2D(64, kernel_size = size, use_bias=False, **kwargs))
model.add(tf.keras.layers.MaxPooling2D((2, 2)))
model.add(tf.keras.layers.BatchNormalization(scale=False))

model.add(lq.layers.QuantConv2D(64, kernel_size = size, use_bias=False, **kwargs))
model.add(tf.keras.layers.BatchNormalization(scale=False))
model.add(tf.keras.layers.Flatten())

model.add(lq.layers.QuantDense(64, use_bias=False, **kwargs))
model.add(tf.keras.layers.BatchNormalization(scale=False))
model.add(lq.layers.QuantDense(10, use_bias=False, **kwargs))
model.add(tf.keras.layers.BatchNormalization(scale=False))
model.add(tf.keras.layers.Activation("softmax"))
tf.keras.utils.plot_model(model,show_shapes=True)

checkpoint_filepath = '/tmp/checkpoint'

checkpoint = ModelCheckpoint(filepath=checkpoint_filepath, monitor='val_loss', 
                              verbose=1, save_best_only=True, mode='min')
es = EarlyStopping(monitor='val_loss', mode='min', verbose=1, patience=30)

#optim = keras.optimizers.Adam(learning_rate=0.1)
lq.models.summary(model)
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])



history = model.fit(train_images, train_labels, 
                    validation_data=(test_images,test_labels),
                    batch_size=64, 
                    epochs=100,
                    callbacks=[es,checkpoint])

model.load_weights(checkpoint_filepath)
test_loss, test_acc = model.evaluate(test_images, test_labels)

print(f"Test accuracy {test_acc * 100:.2f} %")
```
and the results that i obtain are the following:
2 bit and kernel size 2x2 input float: 98.91%
2 bit and kernel size 2x2 input quant: 98.96%
4 bit and kernel size 4x4 input float: 99.18%
4 bit and kernel size 4x4 input quant: 99.26%

I know these results are pretty much the same, but one would expect to have a drop in accuracy if the input is quantized. Or am i wrong?
